lib/srv/server/azure_installer.go

```diff
@@ -45,22 +41,21 @@ type AzureInstaller struct {
 type AzureRunRequest struct {
 	Client          azure.RunCommandClient
 	Instances       []*armcompute.VirtualMachine
-	Params          []string
+	InstallerParams *types.InstallerParams
+	ProxyAddrGetter func(context.Context) (string, error)
 	Region          string
 	ResourceGroup   string
-	ScriptName      string
-	PublicProxyAddr string
-	ClientID        string
-	InstallSuffix   string
-	UpdateGroup     string
-
-	randReader func(b []byte) (n int, err error)
 }
 
 // Run runs a command on a set of virtual machines and then blocks until the
 // commands have completed.
 func (ai *AzureInstaller) Run(ctx context.Context, req AzureRunRequest) error {
-	script, err := getInstallerScript(req)
+	// Azure treats scripts with the same content as the same invocation and
+	// won't run them more than once. This is fine when the installer script
+	// succeeds, but it makes troubleshooting much harder when it fails. To
+	// work around this, we generate a random string and append it as a comment
+	// to the script, forcing Azure to see each invocation as unique.
```

Comment:
Definitely more of a nitpick, but would it make sense to move or duplicate this comment to better associate it with `withNonceComment()`? The connection between what a "nonce comment" is and where it's generated is a bit lost now that the code to generate it has moved elsewhere.